index.js

```diff
@@ -1,82 +1,194 @@
 'use strict';
 
-function constant(value) {
-    return function() {
-        return value;
+module.exports = create;
+module.exports.migrate = migrate;
+module.exports.validate = validate;
+module.exports.is = isScale;
+
+function isScale(value) {
+    return isObject(value);
+}
+
+function migrate(parameters) {
+    parameters = clone(parameters);
+    if (parameters.stops) {
+        parameters.domain = [];
+        parameters.range = [];
+
+        for (var i = 0; i < parameters.stops.length; i++) {
+            parameters.domain.push(parameters.stops[i][0]);
+            parameters.range.push(parameters.stops[i][1]);
+        }
+
+        delete parameters.stops;
     }
+
+    return parameters;
 }
 
-function interpolateNumber(a, b, t) {
-    return (a * (1 - t)) + (b * t);
+function validate(parameters) {
+    if (!isScale(parameters)) return;
+
+    assert(parameters.type, 'Scale must have a type of "power" or "ordinal"');
+    assert(!parameters.property || isString(parameters.property), 'Scale property parameter must be null or a string');
+    assert(parameters.domain, 'Scale must have a domain');
+    assert(parameters.range, 'Scale must have a range');
+    assert(parameters.domain.length === parameters.range.length, "Scale's domain must have the same number of domain elements as range elements");
+    assert(parameters.domain.length > 0, 'Scale must have more than 0 domain elements');
+
+    if (parameters.type === 'power') {
+        assert(!parameters.rounding || parameters.rounding === 'none'  || parameters.rounding === 'floor', 'Scale rounding parameter must be one of "none", or "floor"');
+        assert(!parameters.base || isNumeric(parameters.base), 'Scale base parameter must be null or a number');
+    }
 }
 
-function interpolateArray(a, b, t) {
-    var result = [];
-    for (var i = 0; i < a.length; i++) {
-        result[i] = interpolateNumber(a[i], b[i], t);
+function create(parameters) {
+    validate(parameters);
+
+    // If the scale doesn't define a range, no interpolation will occur and the output value will be
+    // constant.
+    if (!isObject(parameters)) {
+        return function() { return function() { return parameters; }; };
     }
-    return result;
+
+    var property = parameters.property !== undefined ? parameters.property : '$zoom';
+
+    return function(attributes) {
+        var attribute = attributes[property];
+        if (attribute === undefined) {
```

Comment:
Instead of checking whether or not `property` is in the first set of `attributes`, let's check whether or not it begins with `$`.